web/src/pages/Dashboard/index.tsx

```diff
@@ -26,11 +30,49 @@ const ConnectWalletContainer = styled.div`
   color: ${({ theme }) => theme.primaryText};
 `;
 
+const calculatePages = (status: number, data: UserQuery | undefined, casesPerPage: number, myAppeals?: number) => {
+  if (!data) {
+    return 0;
+  }
+
+  let totalPages = 0;
+
+  switch (status) {
+    case 1:
+      totalPages = data.user?.totalDisputes - data.user?.totalResolvedDisputes;
+      break;
+    case 2:
+      totalPages = data.user?.totalResolvedDisputes ?? 0;
+      break;
+    case 3:
+      totalPages = myAppeals ?? 0;
+      break;
+    default:
+      totalPages = data.user?.totalDisputes ?? 0;
+  }
+
+  return totalPages / casesPerPage;
+};
+
 const Dashboard: React.FC = () => {
-  const { isConnected } = useAccount();
+  const { isConnected, address } = useAccount();
+  const { combinedQueryFilters, debouncedSearch, timeFilter, statusFilter } = useFiltersContext();
   const [currentPage, setCurrentPage] = useState(1);
+  const [courtFilter, setCourtFilter] = useState(0);
   const casesPerPage = 3;
-  const { data } = useCasesQuery(casesPerPage * (currentPage - 1));
+  const disputeSkip = debouncedSearch ? 0 : 3 * (currentPage - 1);
+  const direction = timeFilter === 0 ? OrderDirection.Desc : OrderDirection.Asc;
+  const courtChoice = courtFilter === 0 ? {} : { court: courtFilter.toString() };
+  const { data: userAppealCases } = useMyCasesCounterQuery(address, { period: Period.Appeal, ...courtChoice });
```

Comment:
I don't like this approach, I'm going to modify the subgraph to add `numberAppealingDisputes` to the user entity